ethcore/engines/validator-set/src/safe_contract.rs

```diff
@@ -200,14 +211,37 @@ fn prove_initial(contract_address: Address, header: &Header, caller: &Call) -> R
 }
 
 impl ValidatorSafeContract {
-	pub fn new(contract_address: Address) -> Self {
+	pub fn new(contract_address: Address, posdao_transition: Option<BlockNumber>) -> Self {
 		ValidatorSafeContract {
 			contract_address,
 			validators: RwLock::new(MemoryLruCache::new(MEMOIZE_CAPACITY)),
 			client: RwLock::new(None),
+			report_queue: Mutex::new(ReportQueue::default()),
+			resent_reports_in_block: Mutex::new(0),
+			posdao_transition,
+		}
+	}
+
+	fn transact(&self, data: Bytes, nonce: U256) -> Result<(), EthcoreError> {
+		let client = self.client.read().as_ref().and_then(Weak::upgrade).ok_or("No client!")?;
+		let full_client = client.as_full_client().ok_or("No full client!")?;
+
+		let tx_request = TransactionRequest::call(self.contract_address, data).gas_price(U256::zero()).nonce(nonce);
+		match full_client.transact(tx_request) {
+			Ok(()) | Err(transaction::Error::AlreadyImported) => Ok(()),
+			Err(e) => Err(e)?,
 		}
 	}
 
+	pub(crate) fn queue_report(&self, addr: Address, block: BlockNumber, data: Vec<u8>) {
```

Comment:
```suggestion
	pub(crate) fn enqueue_report(&self, addr: Address, block: BlockNumber, data: Vec<u8>) {
```

Reply:
Done in https://github.com/paritytech/parity-ethereum/pull/11245/commits/f7c6325a9a0f7cc9bbe4000a9e6a2778ccb8129f